raster/src/main/scala/geotrellis/raster/ConstantTile.scala

```diff
@@ -81,18 +80,17 @@ abstract class ConstantTile extends Tile {
     * @param   newType  The type of cells that the result should have
     * @return            The new Tile
     */
-  def convert(newType: CellType): Tile = {
+  def convert(newType: CellType): Tile =
     newType match {
       case BitCellType => new BitConstantTile(if (iVal == 0) false else true, cols, rows)
-      case ct: ByteCells => ByteConstantTile(iVal.toByte, cols, rows, ct)
+      case ct: ByteCells => ByteConstantTile(i2b(iVal), cols, rows, ct)
       case ct: UByteCells => UByteConstantTile(iVal.toByte, cols, rows, ct)
-      case ct: ShortCells => ShortConstantTile(iVal.toShort , cols, rows, ct)
-      case ct: UShortCells =>  UShortConstantTile(iVal.toShort , cols, rows, ct)
-      case ct: IntCells =>  IntConstantTile(iVal , cols, rows, ct)
-      case ct: FloatCells => FloatConstantTile(dVal.toFloat , cols, rows, ct)
+      case ct: ShortCells => ShortConstantTile(i2s(iVal), cols, rows, ct)
+      case ct: UShortCells => UShortConstantTile(i2us(iVal) , cols, rows, ct)
+      case ct: IntCells => IntConstantTile(iVal, cols, rows, ct)
+      case ct: FloatCells => FloatConstantTile(d2f(dVal), cols, rows, ct)
       case ct: DoubleCells => DoubleConstantTile(dVal, cols, rows, ct)
```

Comment:
That's the bug actually^ a tiny gotcha that caused chaos.
i.e. `Int.MinValue.toByte / .toShort` does not returns us an expected NoDataValue.